src/unix/fs.c

```diff
@@ -777,6 +772,46 @@ static int uv__fs_fstat(int fd, uv_stat_t *buf) {
 }
 
 
+typedef ssize_t (*uv__fs_buf_iter_yield)(uv_fs_t* req);
+static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_yield yield) {
+  unsigned int iovmax;
+  unsigned int nbufs;
+  uv_buf_t* bufs;
+  ssize_t total;
+  ssize_t result;
+
+  iovmax = uv__getiovmax();
+  nbufs = req->nbufs;
+  bufs = req->bufs;
+  total = 0;
+
+  while (nbufs > 0) {
+    req->nbufs = nbufs;
+    if (req->nbufs > iovmax)
+      req->nbufs = iovmax;
+
+    result = yield(req);
+    if (result < 0) {
```

Comment:
The reason I included the zero, is that a read can return zero for a succesful read, indicating the end of a stream.
If you do not check for zero, another read will be executed, resulting in a failure (reading past the end of a file, or worse, returning zero again.) The same hold for a write were a zero indicates nothing was written, which you don't want to redo in the iteration context.